Assets/Scripts/Scenes/InGame/Stage/StageRack.cs

```diff
@@ -0,0 +1,112 @@
+using Scenes.Ingame.Player;
+using UnityEngine;
+using DG.Tweening;
+
+namespace Scenes.Ingame.Stage
+{
+    public class StageRack : MonoBehaviour, IInteractable
+    {
+        [SerializeField]
+        private DrawType drawType;
+        private enum DrawType
+        {
+            RightOpen, //�E�����ɂ�����
+            LeftOpen, //�������ɂ�����
+            DrawOpen, // �����ĊJ����
+        }
+
+        [SerializeField]
+        private bool _initialStateOpen = true;
+        private bool _isOpen = false;
+        private bool _isAnimation = false;
+        private Vector3 R_OPENVALUE = new Vector3(0, -110, 0);
+        private Vector3 L_OPENVALUE = new Vector3(0, 110, 0);
+        private Vector3 D_OPENVALUE = new Vector3(1, 0, 0);
+        private BoxCollider _rackCollider;
+        public void Intract(PlayerStatus status)
+        {
+            if (Input.GetMouseButtonDown(1) && _isAnimation == false)
+            {
+                _rackCollider.isTrigger = true;
+                _isAnimation = true;
+                switch (_isOpen)
+                {
+                    case true:
+                        RackClose();
+                        _isOpen = false;
+                        break;
+                    case false:
+                        RackOpen();
+                        _isOpen = true;
+                        break;
+                }
+            }
+        }
+
+        void Awake()
+        {
+            _rackCollider = GetComponent<BoxCollider>();
+            if (_initialStateOpen)
+            {
+                _rackCollider.isTrigger = false;
+                RackOpen();
+                _isOpen = true;
+            }
+        }
+        private void RackOpen()
+        {
+            if (drawType == DrawType.RightOpen)
+            {
+                transform.DORotate(R_OPENVALUE, 1).SetRelative(true).SetEase(Ease.InOutSine).OnComplete(() =>
+                {
+                    _rackCollider.isTrigger = false;
+                    _isAnimation = false;
+                });
+            }
+            else if (drawType == DrawType.LeftOpen)
+            {
+                transform.DORotate(L_OPENVALUE, 1).SetRelative(true).SetEase(Ease.InOutSine).OnComplete(() =>
+                {
+                    _rackCollider.isTrigger = false;
+                    _isAnimation = false;
+                });
+            }
+            else if (drawType == DrawType.DrawOpen)
+            {
+                transform.DOMove(0.3f * D_OPENVALUE, 1).SetRelative(true).SetEase(Ease.InOutSine).OnComplete(() =>
+                {
+                    _rackCollider.isTrigger = false;
+                    _isAnimation = false;
+                });
```

Comment:
```suggestion
                transform.DORotate(R_OPENVALUE, 1).SetRelative(true).SetEase(Ease.InOutSine).OnComplete(() =>
                {
                    AnimationcComplete();
                });
            }
            else if (drawType == DrawType.LeftOpen)
            {
                transform.DORotate(L_OPENVALUE, 1).SetRelative(true).SetEase(Ease.InOutSine).OnComplete(() =>
                {
                    AnimationcComplete();
                });
            }
            else if (drawType == DrawType.DrawOpen)
            {
                transform.DOMove(0.3f * D_OPENVALUE, 1).SetRelative(true).SetEase(Ease.InOutSine).OnComplete(() =>
                {
                    AnimationcComplete();
                });
``` 
参考元のStageDoorが少し悪い部分がありますが、今回のような同じ内容の処理が多くある場合は、関数として別に置くと、後からの変更に強くなるため別にした方が良いです。